### ISSUE TITLE
Is this a problem with the bank or I don't understand it?

### ISSUE THREAD

**Issue by 420116199208022715** [opencv/opencv#13342]
## Problems found in preparing for data training

**Here's part of the data**
```
/home/scorpius/Pictures/orl_faces/s1/1.pgm    4
/home/scorpius/Pictures/orl_faces/s1/10.pgm    4
/home/scorpius/Pictures/orl_faces/s1/2.pgm    4
/home/scorpius/Pictures/orl_faces/s1/3.pgm    4
/home/scorpius/Pictures/orl_faces/s1/4.pgm    4
/home/scorpius/Pictures/orl_faces/s1/5.pgm    4
/home/scorpius/Pictures/orl_faces/s1/6.pgm    4
/home/scorpius/Pictures/orl_faces/s1/7.pgm    4
/home/scorpius/Pictures/orl_faces/s1/8.pgm    4
/home/scorpius/Pictures/orl_faces/s1/9.pgm    4
/home/scorpius/Pictures/orl_faces/s10/1.pgm    37
/home/scorpius/Pictures/orl_faces/s10/10.pgm    37
/home/scorpius/Pictures/orl_faces/s10/2.pgm    37
/home/scorpius/Pictures/orl_faces/s10/3.pgm    37
/home/scorpius/Pictures/orl_faces/s10/4.pgm    37
/home/scorpius/Pictures/orl_faces/s10/5.pgm    37
/home/scorpius/Pictures/orl_faces/s10/6.pgm    37
/home/scorpius/Pictures/orl_faces/s10/7.pgm    37
/home/scorpius/Pictures/orl_faces/s10/8.pgm    37
/home/scorpius/Pictures/orl_faces/s10/9.pgm    37
```
**Official guidance**
* inheritance relationship
```
     java.lang.Object
        org.opencv.core.Algorithm
            org.opencv.face.FaceRecognizer 
```
* reference API
```java
public void train(java.util.List<Mat> src,
                  Mat labels)
```

**Below is the code.***
```java
public class DataAcquisition {

    static List<Mat> images = new ArrayList<>();
    static List<Integer> labels = new ArrayList<>(); //

    public static void sequenceFill(String path) throws FileNotFoundException, IOException {
        File file = new File(path);
        BufferedReader br = new BufferedReader(new FileReader(file));
        String line;
        while ((line = br.readLine()) != null) {
            String[] split = line.split(" ");
            images.add(Imgcodecs.imread(split[0]));
            labels.add(Integer.parseInt(split[split.length - 1]));
        }
    }

    public static void main(String[] args) throws FileNotFoundException, IOException {
        System.loadLibrary(Core.NATIVE_LIBRARY_NAME);
        String path = "/home/scorpius/Pictures/orl_faces/images.txt";
        sequenceFill(path);
        LBPHFaceRecognizer model = LBPHFaceRecognizer.create();
        //model.train(images, labels); ??? -_-#
        Mat img = images.get(images.size()-1);
        //int predict_label = model.predict_label(img);//error ??? -_-#
        //System.out.println("检出标记:" + predict_label + " 定义标记:" + labels.get(labels.size()-1));
    }
}
```
* Very confusing, according to the C implementation reference rewrite six months ago, now found that C + + and Java will be stuck in these two places, is it not to pass the int sequence? Is it a separate Mat on the API now to convert a one-dimensional linear sequence into a Mat matrix?

**Comment by 420116199208022715** [comment]
Strangely enough, pull code has been missing Mat objects on Linux in recent days

**Comment by berak** [comment]
this is not a library bug, but a problem with your code.

the FaceRecognizer class expects a `Mat` for the train labels, not a `List<Integer>` 
(there is no possible  conversion between those)

use:

    Mat labels = new Mat();
    ...
    labels.push_back( new MatOfInt(37) );

**Comment by 420116199208022715** [comment]
> 这不是库错误，而是代码问题。
> 
> 所述[FaceRecognizer类](https://docs.opencv.org/master/javadoc/org/opencv/face/FaceRecognizer.html)期望一个`Mat`火车标签，而不是一个`List<Integer>`
> （有那些之间没有可能的转换）
> 
> 使用：
> 
> ```
> Mat labels = new Mat();
> ...
> labels.push_back( new MatOfInt(37) );
> ```

Thank you very much.

**Comment by berak** [comment]
@420116199208022 -- if it solved your problem, please kindly  close the issue ;)
